src/hotspot/share/opto/compile.cpp

```diff
@@ -3666,6 +3666,33 @@ void Compile::final_graph_reshaping_main_switch(Node* n, Final_Reshape_Counts& f
 
   case Op_LoadVector:
   case Op_StoreVector:
+#ifdef ASSERT
+#ifdef AMD64
+    // Add VerifyAlignment node between adr and load / store.
+    if (AlignVector && VerifyAlignVector) {
```

Comment:
`CompilerConfig::check_args_consistency()` seems to be too late. The reason is that `AlignVector` can still be changed by `VM_Version::get_processor_features` because of `UseUnalignedLoadStores`.
@vnkozlov do you have any other suggestion where to do this - ie when all the flags are final?

Reply:
@vnkozlov ok, this seems to work. I now disable `VerifyAlignVector` with a warning.
`Java HotSpot(TM) 64-Bit Server VM warning: VerifyAlignVector disabled because AlignVector not enabled.`